src/Raven.Client/Documents/Operations/Replication/BucketMigrationReplication.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Replication;
+using Raven.Client.ServerWide;
+using Sparrow;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.Replication
+{
+    public class BucketMigrationReplication : ReplicationNode
+    {
+        public int Bucket;
+        public int Shard;
+        public readonly string Node;
+        public readonly long MigrationIndex;
+
+        public BucketMigrationReplication(string node, long migrationIndex)
+        {
+            Node = node ?? throw new ArgumentNullException(nameof(node));
+            MigrationIndex = migrationIndex;
+        }
+
+        public bool ForBucketMigration(BucketMigration migration)
```

Comment:
static? not sure how this is bound to this class?

Reply:
this methods check whether the an incoming bucket migration (the `BucketMigration migration` input) is already handled by this class.